drivers/platform/mbed/mbed_irq.cpp

```diff
@@ -0,0 +1,569 @@
+	case EXTERNAL_INT_ID1:
+	case EXTERNAL_INT_ID2:
+	case EXTERNAL_INT_ID3:
+	case EXTERNAL_INT_ID4:
+	case EXTERNAL_INT_ID5:
+		ext_interrupt = (InterruptIn *)(((struct mbed_irq_desc *)(
+				desc->extra))->int_obj);
+		ext_interrupt->enable_irq();
+		break;
+
+	case TICKER_INT_ID:
+		ticker = (mbed::Ticker *)(((struct mbed_irq_desc *)(desc->extra))->int_obj);
+		ticker->attach(mbed_ticker_id_callback,
+			       microseconds(((struct mbed_irq_desc *)(desc->extra))->ticker_period_usec));
+		break;
+
+	case UART_RX_INT_ID1:
+		uart_rx_port = (mbed::UnbufferedSerial *)(((struct mbed_irq_desc *)(
+					desc->extra))->int_obj);
+		uart_rx_port->attach(mbed_uart_rx_id1_callback, UnbufferedSerial::RxIrq);
+		break;
+
+	default:
+		return -EINVAL;
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief	Disable specific interrupt
+ * @param	desc[in] - The IRQ controller descriptor.
+ * @param	irq_id[in] - Interrupt identifier.
+ * @return	SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t mbed_irq_disable(struct irq_ctrl_desc *desc, uint32_t irq_id)
+{
+	InterruptIn *ext_interrupt;
+	mbed::UnbufferedSerial *uart_rx_port;
+	mbed::Ticker *ticker;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	switch (irq_id) {
+	case EXTERNAL_INT_ID1:
+	case EXTERNAL_INT_ID2:
+	case EXTERNAL_INT_ID3:
+	case EXTERNAL_INT_ID4:
+	case EXTERNAL_INT_ID5:
+		ext_interrupt = (InterruptIn *)(((struct mbed_irq_desc *)(
+				desc->extra))->int_obj);
+		ext_interrupt->disable_irq();
+		break;
+
+	case TICKER_INT_ID:
+		ticker = (mbed::Ticker *)(((struct mbed_irq_desc *)(desc->extra))->int_obj);
+		ticker->detach();
+		break;
+
+	case UART_RX_INT_ID1:
+		uart_rx_port = (mbed::UnbufferedSerial *)(((struct mbed_irq_desc *)(
+					desc->extra))->int_obj);
+		uart_rx_port->attach(NULL, UnbufferedSerial::RxIrq);
+		break;
+
+	default:
+		return -EINVAL;
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief	Free the resources allocated by mbed_irq_ctrl_init()
+ * @param	desc[in, out] - Interrupt controller descriptor.
+ * @return	SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t mbed_irq_ctrl_remove(struct irq_ctrl_desc *desc)
+{
+	uint8_t irq_id;
+	mbed::UnbufferedSerial *uart_rx_port;
+	mbed::Ticker *ticker;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	if (!(((struct mbed_irq_desc *)desc->extra)->int_obj))
+		return -EINVAL;
+
+	/* Unregister all callbacks */
+	for (irq_id = 0; irq_id < (uint8_t)NB_INTERRUPTS; irq_id++) {
+		if (mbed_irq_unregister(desc, irq_id) != SUCCESS)
+			return -EIO;
+	}
+
+	switch (desc->irq_ctrl_id) {
+	case EXTERNAL_INT_ID1:
+	case EXTERNAL_INT_ID2:
+	case EXTERNAL_INT_ID3:
+	case EXTERNAL_INT_ID4:
+	case EXTERNAL_INT_ID5:
+		delete((InterruptIn *)(((struct mbed_irq_desc *)desc->extra)->int_obj));
+		break;
+
+	case TICKER_INT_ID:
+		/* Ticker doesn't have destructor implementation so only detach is used */
+		ticker = (mbed::Ticker *)(((struct mbed_irq_desc *)desc->extra)->int_obj);
+		ticker->detach();
+		break;
```

Comment:
We should `delete(((struct mbed_irq_desc *)desc->extra)->int_obj);` here as well.